pkg/webhook/webhook.go

```diff
@@ -259,6 +259,65 @@ func writeResponse(w http.ResponseWriter, ar *v1beta1.AdmissionReview) {
 	w.Write(resp)
 }
 
+func addVolDownwardAPI(patch []jsonPatchOperation) []jsonPatchOperation {
+	labels := corev1.ObjectFieldSelector{
+		FieldPath:"metadata.labels",
+	}
+	dAPILabels := corev1.DownwardAPIVolumeFile{
+		Path: "labels",
+		FieldRef: &labels,
+	}
+	annotations := corev1.ObjectFieldSelector{
+		FieldPath:"metadata.annotations",
+	}
+	dAPIAnnotations := corev1.DownwardAPIVolumeFile{
+		Path: "annotations",
+		FieldRef: &annotations,
+	}
+	dAPIItems := []corev1.DownwardAPIVolumeFile{dAPILabels, dAPIAnnotations}
+	dAPIVolSource := corev1.DownwardAPIVolumeSource{
+		Items: dAPIItems,
+	}
+	volSource := corev1.VolumeSource{
+		DownwardAPI: &dAPIVolSource,
+	}
+	vol := corev1.Volume{
+		Name: "podinfo",
+		VolumeSource: volSource,
+	}
+
+	patch = append(patch, jsonPatchOperation{
+		Operation: "add",
+		Path:      "/spec/volumes/-", // NOTE: in future we may want to patch specific container (not always the first one)
+		Value:    vol,
+	})
+
+	return patch
+}
+
+func addVolumeMount(patch []jsonPatchOperation) []jsonPatchOperation {
+
+	vm := corev1.VolumeMount{
+		Name: "podinfo",
+		ReadOnly: false,
+		MountPath: "/etc/podinfo",
+	}
+
+	patch = append(patch, jsonPatchOperation{
+		Operation: "add",
+		Path:      "/spec/containers/0/volumeMounts/-", // NOTE: in future we may want to patch specific container (not always the first one)
```

Comment:
We might want to loop over each container in Pod and mount the volume `/etc/podinfo`. The reason for this is that containers in the same Pod don't share the process namespace. meaning container-1 cannot access container-0/etc/podinfo unless `shareProcessNamespace: true` is configured in Pod spec.

Reply:
OK, good input. I can certainly loop over all the containers in the pod, and add the vol and mounts. Will add that to the next patch.

Reply:
Discussed with Zenghui, cleanup of hard-coded containers (including for resource requests and limits) can be addressed in future PR.